saleor/graphql/discount/mutations/sale_create.py

```diff
@@ -90,12 +93,49 @@ def clean_instance(cls, info, instance):
     def perform_mutation(cls, _root, info, **data):
         response = super().perform_mutation(_root, info, **data)
         instance = getattr(response, cls._meta.return_field_name).node
-        current_catalogue = fetch_catalogue_info(instance)
+        cls.send_sale_notifications(info, instance)
+        return response
+
+    @classmethod
+    def send_sale_notifications(cls, info, instance):
+        current_catalogue = convert_catalogue_info_to_global_ids(
+            fetch_catalogue_info(instance)
+        )
 
         transaction.on_commit(
             lambda: info.context.plugins.sale_created(
                 instance,
-                convert_catalogue_info_to_global_ids(current_catalogue),
+                current_catalogue,
             )
         )
-        return response
+
+        manager = info.context.plugins
+        update_fields = []
+        cls.send_sale_started_or_ended_notification(
+            manager, instance, current_catalogue, "start", update_fields
+        )
+        cls.send_sale_started_or_ended_notification(
+            manager, instance, current_catalogue, "end", update_fields
+        )
+        if update_fields:
+            instance.save(update_fields=update_fields)
+
+    @staticmethod
+    def send_sale_started_or_ended_notification(
+        manager, instance, catalogue, field, update_fields
+    ):
+        """Send a notification about starting or ending sale if it hasn't been sent yet.
+
+        Send the notification if the notification field is set to False and the starting
+        or ending date already passed.
+        """
+        now = datetime.now(pytz.utc)
+        notification_field = f"{field}ed_notification_sent"
+        date = getattr(instance, f"{field}_date")
+        if date and date <= now:
+            if field == "start":
+                manager.sale_started(instance, catalogue)
+            else:
+                manager.sale_ended(instance, catalogue)
```

Comment:
[Question] It is SaleCreate mutation. It only creates a new sale. In that case do we need to trigger `sale_ended` hook? Like is it expected that we will send webhook that sale is ended even if there was never a way to use it?

Reply:
Hm, but It's possible to create `Sale` that has started and ended date in the past. And in this case, we will send the notification for the start and not for the end, and products might be on sale because of that.

Reply:
Yeah, but what is the reason for sending such notifications for past events? It is already ended from the beginning of its life and I am not sure if this gives you anything except a noise that you (as an app) have to filter out. (But maybe I don't know all requirments or usecases).

Reply:
I introduced a new solution that handles this case and does not send the notification in such a situation.